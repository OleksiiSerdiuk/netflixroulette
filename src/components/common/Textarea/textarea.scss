.base-textarea {
  display: flex;
  flex-direction: column;

  &__label {
    color: var(--color-carnation);
    width: 100%;
    margin-bottom: 5px;
  }

  &__field {
    width: 100%;
    padding-top: 10px;
    outline: none;
    padding-left: 1rem;
    color: var(--color-white);
    border: none;
    background: rgba(50, 50, 50, 0.948044);
    border-radius: 4px;
    resize: none;
    min-height: 197px;

    &::placeholder {
      font-size: 16px;
      color: var(--color-emperor);
    }
  }
}
