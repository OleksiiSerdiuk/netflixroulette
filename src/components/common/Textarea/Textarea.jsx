import './textarea.scss';
import { ErrorMessage, Field } from 'formik';
import TextError from '../TextError';

const Textarea = (props) => {
  const { label, name, type, ...rest } = props;

  return (
    <label className='base-textarea'>
      <span className='base-textarea__label'>{label}</span>
      <Field
        as='textarea'
        id={name}
        name={name}
        className='base-textarea__field'
        {...rest}
      />
      <ErrorMessage name={name} component={TextError} />
    </label>
  );
};

export default Textarea;
