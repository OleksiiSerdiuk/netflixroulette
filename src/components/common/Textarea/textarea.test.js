import { render } from '@testing-library/react';
import Textarea from './Textarea';
import { Formik } from 'formik';
describe('textarea', () => {
  test('renders textarea snapshot', () => {
    const { asFragment } = render(
      <Formik>
        <Textarea
          id='test'
          control='textarea'
          label='Overview'
          name='overview'
        />
      </Formik>
    );
    expect(asFragment(<Textarea />)).toMatchSnapshot();
  });
});
