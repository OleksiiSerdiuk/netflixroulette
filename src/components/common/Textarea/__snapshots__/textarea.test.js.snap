// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textarea renders textarea snapshot 1`] = `
<DocumentFragment>
  <label
    class="base-textarea"
  >
    <span
      class="base-textarea__label"
    >
      Overview
    </span>
    <textarea
      class="base-textarea__field"
      control="textarea"
      id="test"
      name="overview"
    />
  </label>
</DocumentFragment>
`;
